torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -273,13 +288,18 @@ static void UpdateScalarTypeForOutput(
 
 static void ImplicitCastNodeForONNX(Node* n) {
   if (IsImplicitCastSupported(n->kind())) {
-    auto expected_scalar_type = InferExpectedScalarType(n);
+    std::unordered_map<NodeKind, c10::ScalarType> opUnsupportedType = {};
+    auto expected_scalar_type = InferExpectedScalarType(n, opUnsupportedType);
```

Comment:
I'd suggest avoid modifying `InferExpectedScalarType`. 
Try to design a function to decide if precision-cast is needed, and get you the higher precision dtype based on op kind and `expected_scalar_type`.